plutus-contract/src/Language/Plutus/Contract/Types.hs

```diff
@@ -239,6 +191,14 @@ mapError ::
 mapError f (Contract c) = Contract c' where
   c' = E.handleError @e (raiseUnderN @'[E.Error e'] c) (E.throwError @e' . f)
 
+-- | Turn a contract with error type 'e' and return type 'a' into one with
+--   error type 'Void' (ie. throwing no errors) that returns 'Either e a'
+runError ::
+  forall s e a.
+  Contract s e a
+  -> Contract s Void (Either e a)
```

Comment:
I think it's quite nice, the types tell you exactly what the contract does (we can't get rid of the `Error` effect because `Contract` has a fixed list of effects which includes `Error`)